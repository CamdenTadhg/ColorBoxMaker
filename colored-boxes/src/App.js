import react from 'react'
import './App.css'
import BoxList from './BoxList.js'

function App() {
return (
  <BoxList />
  )
};

export default App

//Write test for creating a new box
//Write test for deleting a box
