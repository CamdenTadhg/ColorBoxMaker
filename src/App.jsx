import react, { useState } from 'react'
import './App.css'
import BoxList from './BoxList.jsx'
import Box from './Box.jsx'

function App() {
return (
  <BoxList />
  )
};

export default App

//Create Box component (div with background color, width, and height)
  //add a button with text of X that removes that specific box
//Create NewBoxForm
  // form specifies box's width, height, and background color
  // Create new Box in Box array
  // Clear input values
//Write smoke test
//Write snapshot test
//Write test for creating a new box
//Write test for deleting a box