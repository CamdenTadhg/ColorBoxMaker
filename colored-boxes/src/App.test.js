import react from "react";
import {fireEvent, render} from '@testing-library/react'
import App from './App.js'

it('should render correctly', function() {
    render(<App />);
});

it('should match the snapshot', function(){
    const {asFragment} = render(<App/>);
    expect(asFragment()).toMatchSnapshot();
});

it('should delete the clicked box', function(){
    const {getByTestId} = render(<App/>);
    const box = getByTestId('blueviolet1821');
    expect(box).toBeInTheDocument();

    fireEvent.click(getByTestId('blueviolet1821button'));
    expect(box).not.toBeInTheDocument();
});

it('should create a new box based on form data', function(){
    const {getByPlaceholderText, getByTestId, queryByTestId} = render(<App/>);
    expect(queryByTestId('blue2410')).not.toBeInTheDocument();
    const colorInput = getByPlaceholderText('color');
    const widthInput = getByPlaceholderText('width');
    const heightInput = getByPlaceholderText('height');
    const form = getByTestId('newBoxForm');
    fireEvent.input(colorInput, {target: {value: 'blue'}});
    fireEvent.input(widthInput, {target: {value: 24}});
    fireEvent.input(heightInput, {target: {value: 10}});
    fireEvent.submit(form);
    expect(getByTestId('blue2410')).toBeInTheDocument();
});
