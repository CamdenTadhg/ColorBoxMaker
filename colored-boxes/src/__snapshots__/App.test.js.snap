// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form
      class="NewBoxForm"
      data-testid="newBoxForm"
    >
      <label
        class="NewBoxForm-label"
        for="color"
      >
        Color:
        <input
          id="color"
          name="color"
          placeholder="color"
          type="text"
          value=""
        />
      </label>
      <label
        class="NewBoxForm-label"
        for="width"
      >
        Width:
        <input
          id="width"
          name="width"
          placeholder="width"
          type="number"
          value="0"
        />
      </label>
      <label
        class="NewBoxForm-label"
        for="height"
      >
        Height:
        <input
          id="height"
          name="height"
          placeholder="height"
          type="number"
          value="0"
        />
      </label>
      <button>
        Add Box
      </button>
    </form>
    <div
      class="Box"
      data-testid="blueviolet1821"
      style="background-color: blueviolet; width: 18vw; height: 21vh;"
    />
    <button
      class="Box-button"
      data-testid="blueviolet1821button"
    >
      X
    </button>
    <div
      class="Box"
      data-testid="darkmagenta1717"
      style="background-color: darkmagenta; width: 17vw; height: 17vh;"
    />
    <button
      class="Box-button"
      data-testid="darkmagenta1717button"
    >
      X
    </button>
    <div
      class="Box"
      data-testid="darkorchid2823"
      style="background-color: darkorchid; width: 28vw; height: 23vh;"
    />
    <button
      class="Box-button"
      data-testid="darkorchid2823button"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
